core/src/Streamly/Internal/Data/Stream/StreamD/Nesting.hs

```diff
@@ -555,6 +555,19 @@ data ConcatUnfoldInterleaveState o i =
 -- of pairing up the streams we just need to go yielding one element from each
 -- stream and storing the remaining streams and then keep doing rounds through
 -- those in a round robin fashion. This would be much like wAsync.
+--
+-- | This does not pair streams like concatPairsWith, instead, it goes through
+-- each stream one by one and yields one element from each stream. After it
+-- goes to the last stream it reverses the traversal to come back to the first
+-- stream yielding elements from each stream on its way back to the first
+-- stream and so on.
+--
+-- >>> input = Stream.fromList [[1,1],[2,2],[3,3],[4,4],[5,5]]
+-- >>> Stream.fold Fold.toList $ Stream.unfoldManyInterleave Unfold.fromList input
+-- [1,2,3,4,5,5,4,3,2,1]
+--
+-- Note that this is order of magnitude more efficient than "concatPairsWith
+-- wSerial"
```

Comment:
Remove this. This is already present in Streamly.Internal.Data.Stream, we do not want duplication.